lib/osf-components/addon/components/open-resource/component.ts

```diff
@@ -0,0 +1,39 @@
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import Intl from 'ember-intl/services/intl';
+
+import { ResourceTypes } from 'ember-osf-web/models/resource';
+import { getBadgeIcon, getBadgeIconDisabled } from 'ember-osf-web/helpers/open-badges-icon-map';
+import Toast from 'ember-toastr/services/toast';
+import RegistrationModel from 'ember-osf-web/models/registration';
+
+interface OpenResourceArgs {
+    registration: RegistrationModel;
+    resourceType: ResourceTypes;
+    hasResource: boolean;
+}
+
+export default class OpenResourceComponent extends Component<OpenResourceArgs> {
+    @service intl!: Intl;
+    @service toast!: Toast;
+
+    constructor(owner: unknown, args: OpenResourceArgs) {
+        super(owner, args);
+    }
+
+    get getSrc() {
+        if (!this.args.resourceType) {
+            return '';
+        }
+
+        try {
+            if (this.args.hasResource === true) {
+                return getBadgeIcon(this.args.resourceType);
+            } else {
+                return getBadgeIconDisabled(this.args.resourceType);
+            }
+        } catch (e) {
+            return this.toast.error(this.intl.t('registries.overview.resources.' + this.args.resourceType + '.error'));
+        }
+    }
```

Comment:
Error catching is good, but I'm curious what exception you expect to get? The two functions are only switch statements, and they don't raise any exceptions on their own. I guess if you passed in a resourceType that wasn't a string, it miiiight complain, but changes are more likely it would just fall to the default case.